🐞 panic for `nil pointer dereference` when `dtm apply`

## Describe the Bug

command:

```sh
./dtm-darwin-arm64 apply -f config.yaml
```

logs:

```sh

2022-03-11 13:11:58 ℹ [INFO]  Apply started.
2022-03-11 13:11:58 ℹ [INFO]  Using dir <.devstream> to store plugins.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x12cf11228]

goroutine 1 [running]:
github.com/merico-dev/stream/pkg/util/github.(*Client).GetRepoDescription(0x140000d12a0)
        github.com/merico-dev/stream/pkg/util/github/repo.go:45 +0x138
github.com/merico-dev/stream/internal/pkg/plugin/reposcaffolding/github/golang.buildReadState(0x140000aff40)
        github.com/merico-dev/stream/internal/pkg/plugin/reposcaffolding/github/golang/read.go:41 +0x194
github.com/merico-dev/stream/internal/pkg/plugin/reposcaffolding/github/golang.Read(0x140001850e0)
        github.com/merico-dev/stream/internal/pkg/plugin/reposcaffolding/github/golang/read.go:26 +0x18c
github.com/merico-dev/stream/cmd/reposcaffolding/github/golang.Plugin.Read({0x0, 0x0}, 0x140001850e0)
        github.com/merico-dev/stream/cmd/reposcaffolding/github/golang/main.go:27 +0x38
github.com/merico-dev/stream/internal/pkg/pluginengine.Read(0x140000ed468)
        github.com/merico-dev/stream/internal/pkg/pluginengine/plugin.go:45 +0xfc
github.com/merico-dev/stream/internal/pkg/pluginengine.changesForApply({0x10474cdb0, 0x140000ba330}, 0x140000ba2b8)
        github.com/merico-dev/stream/internal/pkg/pluginengine/change_helper.go:83 +0x37c
github.com/merico-dev/stream/internal/pkg/pluginengine.getChanges({0x10474cdb0, 0x140000ba330}, 0x140000ba2b8, {0x10460e7c2, 0x5}, 0x0)
        github.com/merico-dev/stream/internal/pkg/pluginengine/change.go:61 +0x1a8
github.com/merico-dev/stream/internal/pkg/pluginengine.GetChangesForApply({0x10474cdb0, 0x140000ba330}, 0x140000ba2b8)
        github.com/merico-dev/stream/internal/pkg/pluginengine/change.go:43 +0x68
github.com/merico-dev/stream/internal/pkg/pluginengine.Apply({0x16bdab8bc, 0xb}, 0x0)
        github.com/merico-dev/stream/internal/pkg/pluginengine/cmd_apply.go:32 +0x214
main.applyCMDFunc(0x104a4a7c0, {0x140000d0c40, 0x0, 0x2})
        github.com/merico-dev/stream/cmd/devstream/apply.go:22 +0x98
github.com/spf13/cobra.(*Command).execute(0x104a4a7c0, {0x140000d0c20, 0x2, 0x2})
        github.com/spf13/cobra@v1.2.1/command.go:860 +0xa04
github.com/spf13/cobra.(*Command).ExecuteC(0x104a4af40)
        github.com/spf13/cobra@v1.2.1/command.go:974 +0x7b8
github.com/spf13/cobra.(*Command).Execute(0x104a4af40)
        github.com/spf13/cobra@v1.2.1/command.go:902 +0x30
main.main()
        github.com/merico-dev/stream/cmd/devstream/main.go:87 +0x30

```

--- a/pkg/util/github/repo.go
+++ b/pkg/util/github/repo.go
@@ -42,6 +42,10 @@ func (c *Client) GetRepoDescription() (*github.Repository, error) {
 		c.Owner,
 		c.Repo)
 
+	if err != nil {
+		return nil, err
+	}
+
 	if repo == nil && resp.StatusCode == http.StatusNotFound {
 		return repo, nil
 	}

